### ISSUE TITLE
[Warrior] Heroic Strike.

### ISSUE THREAD

**Issue by Kaliram** [dalaranwow/dalaran-wow#3925]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Expected behaviour**: 

When the ability is queued, It is supposed to cause your next off-hand swing be considered a special ability, and as such the off-hand swing should not be subject to glancing blows, the base hit chance should be reduced and its damage should be modified by Impale
**Steps to reproduce the problem**:

1. Attack Something
2. Start spamming heroic strike
3. Watch melee's for glancing blows or misses when at special ability hit cap.

**Include proofs for this behaviour**
https://www.youtube.com/watch?v=MaOC5nRUOCU 

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**

http://wotlk.openwow.com/spell=47450

**Include Screenshots from the issue if necessary**

**Comment by Bonham01** [comment]
I don't know if it is a typo in your post, but heroic strike doesn't affect your offhand swing in any way. Your second link actually confirms this. If you are posting about main hand swings, then maybe but I haven't seen any glancing heroic strikes in my recount/ data logs from raids.

**Comment by robinsch** [comment]
I will mark this as invalid at first.

**Comment by Kaliram** [comment]
Oops, was wrong about the glancing blows part.
I am referring to a bug that existed on retail, that existed on retail that effected off-hand swings as i said. Many other servers have this scripted, and it is a sizable dps increase for warriors( around 10%)


As you can see from the video, its very hard to actually see this, but just try and count the misses.. If the warrior had 5% hit with precision, his offhand swing chance would be around 19% and there would be many, many more misses. Also the Bug does not mean ALL offhand swings will cause the bug to occur, only those directly after a heroic strike.

http://worldoflogs.com/reports/lkfgldp7rx8dbqan/details/25/?s=2512&e=2669#
http://worldoflogs.com/reports/cda48r30qhgaatf6/details/34/?s=6776&e=6962
The percentages on their misses on melee are far lower than they are on this server.

More proof it existed: http://www.mmo-champion.com/threads/732949-landsouls-spreadsheet
http://www.mmo-champion.com/threads/866092-White-critical-strikes-and-the-attack-table?p=10703474&viewfull=1#post10703474
http://blue.mmo-champion.com/topic/20036-warrior-dps-bugs/

**Comment by Kaliram** [comment]
After searching all mmo-champion posts between 6/10 and 12/10 I have found nothing to suggest it was ever fixed.

**Comment by Golganeth** [comment]
Won't fix, not planning to implement bugs/exploits in the game, regardless if they are blizzlike or not. Actual bugs will always have priority.

Closing for now.

**Comment by Aizenscar** [comment]
@Golganeth Dalaran strives to be as blizzlike as possible doesn't it? How would you achieve that goal if you refuse to implement retail features. How is something that existed in retail and boosted a warrior's dps, making them a top dps instead of a middle of the pack dps, by nearly 10% not an actual "bug".

**Comment by robinsch** [comment]
Commit Hash: eb4b7a5e2252a0e70a83d4d5c5c18757f7920eef
Core/Unit: Implemented miss mechanic while having a on next swing attack queued

But it's up to @Lothloryen if it will go on live.

**Comment by TeridanDW** [comment]
Ok, if I understood this correctly this commit:

> Core/Unit: Implemented miss mechanic while having a on next swing attack queued

is suppose to implement a miss chance for off hand swing while we have Heroic Strike qued up, **which in terms helps the warriors to not be "rage starved"** (which is considered to be retail bug).

If this is true after few tests on ptr, I found that I still have 100% hit chance on those attacks. Screenshot below shows me spamming Heroic Strike (all autoattacks are from off hand weapon).
![wowscrnshot_101417_094523](https://user-images.githubusercontent.com/6797611/31573458-7afb4d14-b0c5-11e7-8666-33519fbdbe1a.jpg)

**Comment by robinsch** [comment]
No, this commit is supposed to set your offhand base miss chance to 0 while having Heroic Strike / Cleave queued.

> If this is true after few tests on ptr, I found that I still have 100% hit chance on those attacks. 

And this is the correct behaviour.

**Comment by TeridanDW** [comment]
Ok, then the commit works as intended on PTR.
However I do aggre with @Golganeth:

> Not planning to implement bugs/exploits in the game, regardless if they are blizzlike or not.

Up to @Lothloryen to decide here.

**Comment by tripleslash** [comment]
Hope HoP bug was fixed as well, otherwise u just made warriors insanely OP.

**Comment by TeridanDW** [comment]
Will be with #4365
